src/test/java/io/github/fileanalysissuite/adaptersdk/impls/jaxrs/AdapterSDKTest.java

```diff
@@ -0,0 +1,143 @@
+    public static RetrieveFileListRequest retrieveFileListRequest;
+    public static RetrieveFileDataRequest retrieveFileDataRequest;
+
+    @BeforeClass
+    public static void setup()
+    {
+        final RepositorySettingDefinition repositorySettingDefinition
+            = ConvenientRepositorySettingDefinition.create("MARK", TypeCode.TEXT, true, true);
+
+        adapterDescriptor = ConvenientAdapterDescriptor.create("TEST", repositorySettingDefinition);
+
+        final Map<String, String> configurationOptions = new HashMap<>();
+        configurationOptions.put("a", "b");
+        configurationOptions.put("c", "d");
+
+        final Map<String, String> repositoryOptions = new HashMap<>();
+        repositoryOptions.put("e", "f");
+        repositoryOptions.put("g", "h");
+
+        final RepositoryProperties repositoryProperties
+            = new RepositoryProperties().configurationOptions(configurationOptions).repositoryOptions(repositoryOptions);
+
+        retrieveFileListRequest
+            = new RetrieveFileListRequest().additionalFilter("additionalFilter").repositoryProperties(repositoryProperties);
+
+        final ItemMetadata itemMetadata = new ItemMetadata()
+            .accessedTime("accessedTime")
+            .createdTime("createdTime")
+            .itemLocation("itemLocation")
+            .modifiedTime("modifiedTime")
+            .name("name")
+            .size(Long.MIN_VALUE)
+            .title("title")
+            .version(Integer.SIZE);
+
+        final RepositoryItem repositoryItem = new RepositoryItem().itemId("itemId").metadata(itemMetadata);
+
+        retrieveFileDataRequest
+            = new RetrieveFileDataRequest().repositoryProperties(repositoryProperties).items(Arrays.asList(repositoryItem));
+    }
+
+    @Test
+    public void testAdapterDescriptorGet()
+    {
+        Mockito.when(adapter.createDescriptor()).thenReturn(adapterDescriptor);
+
+        final Response response = target("/adapterDescriptor").request().get();
+        final JsonObject jsonResponse = JsonParser.parseString(response.readEntity(String.class)).getAsJsonObject();
+
+        assertTrue(jsonResponse.size() == 2);
+        assertTrue("TEST".equals(jsonResponse.get("adapterType").getAsString()));
+
+        final JsonArray propertyDefinition = jsonResponse.getAsJsonArray("propertyDefinition");
+        assertTrue(propertyDefinition.size() == 1);
+
+        final JsonObject propertyDefinitionVals = propertyDefinition.get(0).getAsJsonObject();
+
+        assertTrue("MARK".equals(propertyDefinitionVals.get("name").getAsString()));
+        assertTrue(propertyDefinitionVals.get("isRequired").getAsBoolean() == true);
+        assertTrue("true".equals(propertyDefinitionVals.get("isEncrypted").getAsString()));
+        assertTrue(response.getStatus() == 200);
+    }
+
+    @Test
+    public void testRetrieveFileListPost()
+    {
+        Mockito.doNothing().when(adapter).retrieveFileList(any(), any(), any());
```

Comment:
Currently, both `testRetrieveFileListPost` and `testRetrieveFilesDataPost` do not actually test the code in [retrieveFileListPost](https://github.com/FileAnalysisSuite/adaptersdk-impl-jaxrs/blob/US524005/src/main/java/io/github/fileanalysissuite/adaptersdk/impls/jaxrs/AdapterApiImpl.java#L51) or [retrieveFilesDataPost](https://github.com/FileAnalysisSuite/adaptersdk-impl-jaxrs/blob/US524005/src/main/java/io/github/fileanalysissuite/adaptersdk/impls/jaxrs/AdapterApiImpl.java#L65)

I don't see how this can be done if the adapter is being mocked? I found [FileSystemAdapter](https://github.com/FileAnalysisSuite/restadapter-filesystem/blob/237cc5c5da52d732ac9d844b95d301a085c9eb08/restadapter-filesystem-core/src/main/java/io/github/fileanalysissuite/restadapters/filesystem/core/FileSystemAdapter.java) which could be used instead of mocking but maybe this defeats the purpose?